core/src/main/scala/org/apache/spark/deploy/security/HiveDelegationTokenProvider.scala

```diff
@@ -92,8 +93,8 @@ private[security] class HiveDelegationTokenProvider
         s"$principal at $metastoreUri")
 
       doAsRealUser {
-        val hive = Hive.get(conf, classOf[HiveConf])
```

Comment:
Could you be more specific about the scope of this PR in the title and description?

Although the title is `Avoid Hive.get() to compatible with different Hive metastore`,

1. This file still has `Hive.get` at [line 246](https://github.com/apache/spark/pull/20866/files#diff-6fd847124f8eae45ba2de1cf7d6296feL246).
2. `getDelegationTokenFromMetaStore` of `CLIService.java` also has [Hive.get](https://github.com/apache/hive/blob/rel/release-2.3.2/service/src/java/org/apache/hive/service/cli/CLIService.java#L575).

Reply:
Thanks @dongjoon-hyun, seems `RetryingMetaStoreClient` is a better choice and I will try.

Reply:
1. This [`Hive.get()`](https://github.com/apache/spark/blob/v2.3.0/sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala#L239) is different from others, It loaded by  `IsolatedClientLoader`.
2. I can not start a `HiveThriftServer2` in a kerberized cluster, so I'm not sure `CLIService.java` should be updated, How about update it later?

Reply:
Ya. What I asked is the following.
> Could you be more specific about the scope of this PR in the title and description?